R/rdeephaven/R/update_by_ops_wrapper.R

```diff
@@ -14,76 +152,390 @@ UpdateByOp <- R6Class("UpdateByOp",
 #' Default is to compute the cumulative sum for all non-grouping columns.
-#' @return UpdateByOp to be used in `update_by()`.
+#' @return `UpdateByOp` to be used in a call to `update_by()`.
+#'
+#' @examples
+#' \dontrun{
+#' library(rdeephaven)
+#'
+#' # connecting to Deephaven server
+#' client <- Client$new("localhost:10000", auth_type="psk", auth_token="my_secret_token")
+#'
+#' # create data frame, push to server, retrieve TableHandle
+#' df <- data.frame(
+#'   timeCol = seq.POSIXt(as.POSIXct(Sys.Date()), as.POSIXct(Sys.Date() + 0.01), by = "1 sec")[1:500],
+#'   boolCol = sample(c(TRUE,FALSE), 500, TRUE),
+#'   col1 = sample(10000, size = 500, replace = TRUE),
+#'   col2 = sample(10000, size = 500, replace = TRUE),
+#'   col3 = 1:500
+#' )
+#' th <- client$import_table(df)
+#'
+#' # compute cumulative sum of col1 and col2
+#' th1 <- th$
+#'   update_by(uby_cum_sum(c("col1CumSum = col1", "col2CumSum = col2")))
+#'
+#' # compute cumulative sum of col1 and col2, grouped by boolCol
+#' th2 <- th$
+#'   update_by(uby_cum_sum(c("col1CumSum = col1", "col2CumSum = col2")), by="boolCol")
+#'
+#' # compute cumulative sum of col1 and col2, grouped by boolCol and parity of col3
+#' th3 <- th$
+#'   update("col3Parity = col3 % 2")$
+#'   update_by(uby_cum_sum(c("col1CumSum = col1", "col2CumSum = col2")), by=c("boolCol", "col3Parity"))
+#'
+#' client$close()
+#' }
+#'
 #' @export
 uby_cum_sum <- function(cols = character()) {
   verify_string("cols", cols, FALSE)
   return(UpdateByOp$new(INTERNAL_cum_sum(cols)))
 }
 
+#' @name
+#' uby_cum_prod
+#' @title
+#' Cumulative product of specified columns by group
+#' @md
+#'
 #' @description
 #' Creates a cumulative product UpdateByOp that computes the cumulative product of each column in `cols` for each aggregation group.
+#'
+#' @details
+#' This function acts on aggregation groups specified with the `by` parameter of the `update_by()` caller function.
+#' The aggregation groups are defined by the unique combinations of values in the `by` columns. For example,
+#' if `by = c("A", "B")`, then the aggregation groups are defined by the unique combinations of values in the
+#' `A` and `B` columns.
+#'
+#' This function, like other Deephaven `uby` functions, is a generator function. That is, its output is another
+#' function called an [`UpdateByOp`][UpdateByOp] intended to be used in a call to `update_by()`. This detail is typically
+#' hidden from the user. However, it is important to understand this detail for debugging purposes, as the output of
+#' a `uby` function can otherwise seem unexpected.
+#'
 #' @param cols String or list of strings denoting the column(s) to operate on. Can be renaming expressions, i.e. “new_col = col”.
 #' Default is to compute the cumulative product for all non-grouping columns.
-#' @return UpdateByOp to be used in `update_by()`.
+#' @return `UpdateByOp` to be used in a call to `update_by()`.
```

Comment:
`update_by()` is listed multiple times in here without a link.  Seems like it should have a link.  Change probably needed in other places as well.